### PR TITLE
Use on the fly samples generated by backend in feature slice visualizer

### PR DESCRIPTION
**Pull request by timediv**

### CODE REVIEW

**Comment by timediv** [comment]
Okay ready to review

File: assets/js/directives/featureSliceVisualizer.coffee [hunk 1]
```diff
@@ -64,33 +65,24 @@ app.directive 'featureSliceVisualizer', [
...
-            promises = scope.selectedFeatures.map retrieveSamples
-            promises.push retrieveSamples scope.targetFeature
-            promises.push retrieveMarginalDistribution()
-            $q.all promises
-              # TODO remove this annoying timeout
+            retrieveMarginalDistribution()
+              # TODO remove timeout, it waits until layouting is done
               .then -> $timeout null, 1500
               .then -> scope.setupCharts()
               .then -> scope.updateCharts()
```
Comment (danthe96) [inline assets/js/directives/featureSliceVisualizer.coffee]: Can be simplified to `.then scope.setupCharts` (and `.then scope.updateCharts`)

**Comment by timediv** [comment]
yes it's tested and implemented in the backend

File: assets/js/directives/featureSliceVisualizer.coffee (hunk 1 above)
Comment (timediv) [inline assets/js/directives/featureSliceVisualizer.coffee]: You're right.
But that would call SetupCharts with the timeout cancel fnc and updateChart with the return value of setupCharts.
This is more explicit and less prone to errors I think.

Comment (danthe96) [inline assets/js/directives/featureSliceVisualizer.coffee]: Ah ok, didn't know arguments would be chained automatically.

**Pull request merged**